### PR TITLE
Changing BPS shutdown order to shut down ode scheduler first

### PR DESCRIPTION
**Pull request by vinok88**
Registering WaitBeforeShutdownObserver service to shutdown ode scheduler first and taking ode scheduler to the top in the BPELServerImpl.shutdown() method. Resolves the issue -https://wso2.org/jira/browse/BPS-470.

### CODE REVIEW

File: components/bpel/org.wso2.carbon.bpel/src/main/java/org/wso2/carbon/bpel/core/ode/integration/BPELSchedulerShutDown.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
+/**
+ *
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
```
Comment (suhothayan) [inline components/bpel/org.wso2.carbon.bpel/src/main/java/org/wso2/carbon/bpel/core/ode/integration/BPELSchedulerShutDown.java]: Please use the WSO2 License file

File: components/bpel/org.wso2.carbon.bpel/src/main/java/org/wso2/carbon/bpel/core/ode/integration/BPELSchedulerShutDown.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        Scheduler scheduler = ((BPELServerImpl) BPELServiceComponent.getBPELServer()).getScheduler();
+
+        if (scheduler != null) {
+            try {
+                if (log.isDebugEnabled()) {
+                    log.debug("Shutting down quartz scheduler.");
+                }
+                scheduler.shutdown();
+            } catch (Exception e) {
+                log.warn("Scheduler couldn't be shut down.", e);
```
Comment (suhothayan) [inline components/bpel/org.wso2.carbon.bpel/src/main/java/org/wso2/carbon/bpel/core/ode/integration/BPELSchedulerShutDown.java]: Exception message it wrong- Use something like Exception is shutting down scheduler

Comment (vinok88) [inline components/bpel/org.wso2.carbon.bpel/src/main/java/org/wso2/carbon/bpel/core/ode/integration/BPELSchedulerShutDown.java]: This try catch no longer exist in the code. We have removed it later. Thank you.

File: components/bpel/org.wso2.carbon.bpel/src/main/java/org/wso2/carbon/bpel/core/ode/integration/BPELSchedulerShutDown.java (hunk 1 above)
Comment (vinok88) [inline components/bpel/org.wso2.carbon.bpel/src/main/java/org/wso2/carbon/bpel/core/ode/integration/BPELSchedulerShutDown.java]: Fixed in https://github.com/wso2/carbon-business-process/pull/227. Thank you.

**Pull request merged**